src/Configurator/AddLinesConfigurator.php

```diff
@@ -0,0 +1,196 @@
+            }
+            $value = $patch['content'];
+
+            $this->unPatchFile($file, $value);
+        }
+    }
+
+    public function update(RecipeUpdate $recipeUpdate, array $originalConfig, array $newConfig): void
+    {
+        $originalConfig = array_filter($originalConfig, function ($item) {
+            return !isset($item['requires']) || $this->isPackageInstalled($item['requires']);
+        });
+        $newConfig = array_filter($newConfig, function ($item) {
+            return !isset($item['requires']) || $this->isPackageInstalled($item['requires']);
+        });
+
+        $filterDuplicates = function (array $sourceConfig, array $comparisonConfig) {
+            $filtered = [];
+            foreach ($sourceConfig as $sourceItem) {
+                $found = false;
+                foreach ($comparisonConfig as $comparisonItem) {
+                    if ($sourceItem['file'] === $comparisonItem['file'] && $sourceItem['content'] === $comparisonItem['content']) {
+                        $found = true;
+                        break;
+                    }
+                }
+                if (!$found) {
+                    $filtered[] = $sourceItem;
+                }
+            }
+
+            return $filtered;
+        };
+
+        // remove any config where the file+value is the same before & after
+        $filteredOriginalConfig = $filterDuplicates($originalConfig, $newConfig);
+        $filteredNewConfig = $filterDuplicates($newConfig, $originalConfig);
+
+        $this->unconfigure($recipeUpdate->getOriginalRecipe(), $filteredOriginalConfig, $recipeUpdate->getLock());
+        $this->configure($recipeUpdate->getNewRecipe(), $filteredNewConfig, $recipeUpdate->getLock());
+    }
+
+    private function patchFile(string $file, string $value, string $position, ?string $target)
+    {
+        $fileContents = file_get_contents($file);
+
+        if (false !== strpos($fileContents, $value)) {
+            return; // already includes value, skip
+        }
+
+        switch ($position) {
+            case self::POSITION_BOTTOM:
+                $fileContents .= "\n".$value;
+
+                break;
+            case self::POSITION_TOP:
+                $fileContents = $value."\n".$fileContents;
+
+                break;
+            case self::POSITION_AFTER_TARGET:
+                $lines = explode("\n", $fileContents);
+                $targetFound = false;
+                foreach ($lines as $key => $line) {
+                    if (false !== strpos($line, $target)) {
+                        array_splice($lines, $key + 1, 0, $value);
+                        $targetFound = true;
+
+                        break;
+                    }
+                }
+                $fileContents = implode("\n", $lines);
+
+                if (!$targetFound) {
+                    // TODO warn?
```

Comment:
display a message? Could not add lines after "%s" as no such string was found in "%s": $the-lines

Reply:
Good catch! It looks like this now both here and if the file is missing (with wording differences, of course):

<img width="1124" alt="Screenshot 2023-05-22 at 6 37 10 AM" src="https://github.com/symfony/flex/assets/121003/c79140e1-dbdf-4ae7-8b8f-84c471ec9221">

The user would only see this if `warn_if_missing: true` in `manifest.json` (or, of course, if they have verbosity high).